app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -489,15 +490,15 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Activit
                 )
 
                 // Don't show the dialog if we aren't in the tab that started the download
-                if (downloadState.sessionId == sessionManager.selectedSession?.id) {
+                if (downloadState.sessionId == store.state.selectedTabId) {
                     dynamicDownloadDialog.show()
                     browserToolbarView.expand()
                 }
             }
         }
 
         resumeDownloadDialogState(
-            sessionManager.selectedSession?.id,
+            store.state.selectedTabId,
```

Comment:
This is actually a bug here which is fixed by: https://github.com/mozilla-mobile/fenix/pull/18057